src/Tools/dotnet-user-jwts/src/Commands/ListCommand.cs

```diff
@@ -19,50 +19,64 @@ public static void Register(ProjectCommandLineApplication app)
                 Resources.ListCommand_ShowTokenOption_Description,
                 CommandOptionType.NoValue);
 
+            var outputOption = cmd.Option(
+                "-o|--output",
+                Resources.ListCommand_OutputOption_Description,
+                CommandOptionType.SingleValue);
+
             cmd.HelpOption("-h|--help");
 
             cmd.OnExecute(() =>
             {
-                return Execute(cmd.Reporter, cmd.ProjectOption.Value(), showTokensOption.HasValue());
+                return Execute(cmd.Reporter, cmd.ProjectOption.Value(), showTokensOption.HasValue(), outputOption.Value());
             });
         });
     }
 
-    private static int Execute(IReporter reporter, string projectPath, bool showTokens)
+    private static int Execute(IReporter reporter, string projectPath, bool showTokens, string outputFormat)
     {
         if (!DevJwtCliHelpers.GetProjectAndSecretsId(projectPath, reporter, out var project, out var userSecretsId))
         {
             return 1;
         }
         var jwtStore = new JwtStore(userSecretsId);
 
-        reporter.Output(Resources.FormatListCommand_Project(project));
-        reporter.Output(Resources.FormatListCommand_UserSecretsId(userSecretsId));
-
         if (jwtStore.Jwts is { Count: > 0 } jwts)
         {
-            var table = new ConsoleTable(reporter);
-            table.AddColumns(Resources.JwtPrint_Id, Resources.JwtPrint_Scheme, Resources.JwtPrint_Audiences, Resources.JwtPrint_IssuedOn, Resources.JwtPrint_ExpiresOn);
-
-            if (showTokens)
+            switch (outputFormat)
             {
-                table.AddColumns(Resources.JwtPrint_Token);
-            }
+                case "json":
+                    reporter.Output(JsonSerializer.Serialize(jwts, new JsonSerializerOptions { WriteIndented = true }));
```

Comment:
Can a source-generated JSON serializer work here? Makes it easier to enable native AOT or trimming on this tool later.

Reply:
I'd leave that up to @captainsafia as well I'd think if it would, the `create` output would need that as well.

Reply:
Yeah, if need be we can configure a `JsonContext` for the `Jwt` types that are used here.

BTW, @timheuer I have a branch with this change that I was planning on using for the PR if that works for you.

Reply:
Works for me, I was playing with some things and figured I'd submit one, but I figured you already had it lurking somewhere ;-)